src/Services/PluginInfoService.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Findologic\Services;
+
+use Plenty\Modules\Plugin\Contracts\PluginRepositoryContract;
+use Plenty\Modules\Plugin\Models\Plugin;
+use Plenty\Modules\Plugin\PluginSet\Contracts\PluginSetRepositoryContract;
+use Plenty\Plugin\CachingRepository;
+
+class PluginInfoService
+{
+    const PLUGIN_VERSION_CACHE_KEY_PREFIX = 'plugin_version_';
+
+    /**
+     * @var PluginRepositoryContract|null
+     */
+    private $pluginRepository;
+
+    /**
+     * @var CachingRepository
+     */
+    private $cache;
+
+    /**
+     * @var Plugin|null[]
+     */
+    private $plugins = [];
+
+    /**
+     * @return string|null
+     */
+    public function getPluginVersion(string $pluginName)
+    {
+        $cachedVersion = $this->getCache()->get(self::PLUGIN_VERSION_CACHE_KEY_PREFIX . $pluginName);
+        if ($cachedVersion !== null) {
+            return $cachedVersion;
+        }
+
+        if (!$plugin = $this->getPlugin($pluginName)) {
+            return null;
+        }
+
+        if (!$plugin->versionProductive || $plugin->versionProductive == '0.0.0') {
```

Comment:
Maybe put `0.0.0` into a separate constant e.g. `DEFAULT_VERSION` or similar. Maybe you can come off with a better name.